integration-tests/src/test/java/com/example/IntegrationTestRule.java

```diff
@@ -0,0 +1,44 @@
+package com.example;
+
+import org.junit.AssumptionViolatedException;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+final class IntegrationTestRule implements TestRule {
+  private final Backend backend;
+
+  IntegrationTestRule(Backend backend) {
+    this.backend = backend;
+  }
+
+  @Override
+  public Statement apply(Statement base, Description description) {
+    boolean ignoreCodegen = description.getAnnotation(IgnoreCodegen.class) != null;
+    if (ignoreCodegen && backend == Backend.CODEGEN) {
+      return new Statement() {
+        @Override
+        public void evaluate() {
+          throw new AssumptionViolatedException("Ignored in code gen backend");
```

Comment:
Could/should we do the same thing as the `@ReflectBug` here and run the test anyways and check if it passes?

Reply:
It's not as useful since we don't actually run the Dagger compiler on them. I was actually thinking of splitting them out into their own test since they deal exclusively with error conditions that otherwise happen at compile-time.